helix-core/src/main/java/org/apache/helix/api/rebalancer/constraint/dataprovider/CapacityProvider.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.helix.api.rebalancer.constraint.dataprovider;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * An interface for getting participant capacity information.
+ */
+public interface CapacityProvider {
+
+  /**
+   * @param participant
+   * @return The total participant capacity.
+   */
+  int getParticipantCapacity(String participant);
```

Comment:
This API for total capacity. Since we need to calculate the percentage of usage for prioritizing, we need both total capacity and current usage.
Given this, the second option looks better. Thanks for the comments. I will make the change.